runtime/src/deterministic.rs

```diff
@@ -339,6 +333,24 @@ impl Auditor {
         *hash = hasher.finalize().to_vec();
     }
 
+    fn register(&self, name: &str, help: &str) {
+        let mut hash = self.hash.lock().unwrap();
+        let mut hasher = Sha256::new();
+        hasher.update(&*hash);
+        hasher.update(b"register");
+        hasher.update(name.as_bytes());
+        hasher.update(help.as_bytes());
+        *hash = hasher.finalize().to_vec();
```

Comment:
what's going on here? Don't see this hash getting used.

Reply:
Oh I see it's just a locked version of self.hahs, nm.

Reply:
This is how we update the digest in-place in deterministic 👍